provider/provider.go

```diff
@@ -89,17 +89,28 @@ func (p *ProviderCondition) Evaluate(log logr.Logger, ctx map[string]interface{}
 }
 
 func templateCondition(condition []byte, ctx map[string]interface{}) ([]byte, error) {
-	//TODO(fabianvf): this delim thing is a pretty gross hack
-	t, err := template.New("condition").Delims("'{{", "}}'").Parse(string(condition))
-	if err != nil {
-		return nil, err
-	}
-	buf := &bytes.Buffer{}
-	err = t.Execute(buf, ctx)
+	//TODO(shanw-hurley):
+	// this is needed because for the initial yaml read, we convert this to a string,
+	// then when it is used here, we need the value to be whatever is in the context and not
+	// a string nested in the type.
+	// This may require some documentation, but I believe that it should be fine.
+	// example:
+	// xml:
+	//   filepaths: '{{poms.filepaths}}'
+	//    xpath: //dependencies/dependency
+	// converted to
+	// xml:
+	//   filepaths: {{poms.filepaths}}
+	//   xpath: //dependencies/dependency
+	s := strings.ReplaceAll(string(condition), `'`, "")
```

Comment:
You may want to replace the bare `'` with `'{{ -> {{`,  `"{{ -> {{`, `}}' -> }}` and `}}" -> }}` to allow for the possibility of quotes within a condition

Reply:
Updated